lib/shared/inference/function.rb

```diff
@@ -12,7 +12,21 @@ def initialize(name:, description:, type: nil)
         @parameters = []
       end
 
-      def add_parameter(name:, type:, description:, enum: nil, required: false)
+      def add_parameter(parameter = nil, **kwargs)
```

Comment:
Heh this is a nice way of "overloading" ;)

Reply:
lol you asked for it :)